pkg/sql/plan2/function/buitins.go

```diff
@@ -0,0 +1,73 @@
+// Copyright 2021 - 2022 Matrix Origin
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package function
+
+import (
+	"github.com/matrixorigin/matrixone/pkg/container/types"
+	"github.com/matrixorigin/matrixone/pkg/container/vector"
+	"github.com/matrixorigin/matrixone/pkg/pb/plan"
+	"github.com/matrixorigin/matrixone/pkg/vm/process"
+)
+
+func initBuiltIns() {
+	var err error
+
+	for name, fs := range builtins {
+		for _, f := range fs {
+			err = appendFunction(name, f)
+			if err != nil {
+				panic(err)
+			}
+		}
+	}
+}
+
+var builtins = map[string][]Function{
+	"case": {
```

Comment:
Now I agree with people, using string literal is a bad idea -- can you switch plan2/function.go, all the function names from string to an enum or const?

Reply:
I also think using enum is a good choice

Reply:
> 

that is ok. using const int or an enum can help us to find the function by index when we are executing the plan.
I will register the function according to function_id.
But I could not register all the function in plan2/function.go at version 0.5
and only register a few functions at this pull request.